### PR TITLE
Add helper for validating and parsing dtos in serviceproviders.

### PR DESCRIPTION
**Pull request by seilc1**

### CODE REVIEW

File: broker/src/test/groovy/com/swisscom/cloud/sb/broker/util/DtoValidationHelperSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+        when:
+        def result = DtoValidationHelper.deserializeAndValidate(json, NotNullClass.class)
+
+        then:
+        noExceptionThrown()
+    }
+
+    void "Validation for Range is okay"()
+    {
```
Comment (ampersand8) [inline broker/src/test/groovy/com/swisscom/cloud/sb/broker/util/DtoValidationHelperSpec.groovy]: For consistency I'd suggest to have the `{` at the end of the previous line.

File: broker/src/test/groovy/com/swisscom/cloud/sb/broker/util/DtoValidationHelperSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+        when:
+        def result = DtoValidationHelper.deserializeAndValidate(json, ValidationTest.class)
+
+        then:
+        noExceptionThrown()
+    }
+
+    void "Validation for Range is throws if the few elements are present"()
+    {
```
Comment (ampersand8) [inline broker/src/test/groovy/com/swisscom/cloud/sb/broker/util/DtoValidationHelperSpec.groovy]: For consistency I'd suggest to have the `{` at the end of the previous line.

File: broker/src/test/groovy/com/swisscom/cloud/sb/broker/util/DtoValidationHelperSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+        when:
+        def result = DtoValidationHelper.deserializeAndValidate(json, ValidationTest.class)
+
+        then:
+        thrown(ServiceBrokerException)
+    }
+
+    void "Validation for Range is doesn't throw if is null"()
+    {
```
Comment (ampersand8) [inline broker/src/test/groovy/com/swisscom/cloud/sb/broker/util/DtoValidationHelperSpec.groovy]: For consistency I'd suggest to have the `{` at the end of the previous line.

File: broker/src/main/groovy/com/swisscom/cloud/sb/broker/util/DtoValidationHelper.groovy [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import com.fasterxml.jackson.databind.exc.InvalidFormatException
+import com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException
+import com.swisscom.cloud.sb.broker.error.ErrorCode
+import groovy.transform.CompileStatic
+
+import javax.validation.ConstraintViolation
+import javax.validation.Validation
+
+@CompileStatic
+class DtoValidationHelper {
```
Comment (ampersand8) [inline broker/src/main/groovy/com/swisscom/cloud/sb/broker/util/DtoValidationHelper.groovy]: Can optional fields be omitted?

Comment (seilc1) [reply broker/src/main/groovy/com/swisscom/cloud/sb/broker/util/DtoValidationHelper.groovy]: Yes - Added addtional test to proof it (and document it)

Comment (ampersand8) [reply broker/src/main/groovy/com/swisscom/cloud/sb/broker/util/DtoValidationHelper.groovy]: Perfect :+1:

**Pull request merged**